examples/mic_vad_streaming/mic_vad_streaming.py

```diff
@@ -1,41 +1,77 @@
 import time, logging
 from datetime import datetime
 import threading, collections, queue, os, os.path
-import wave
+import deepspeech
+import numpy as np
 import pyaudio
+import wave
 import webrtcvad
 from halo import Halo
-import deepspeech
-import numpy as np
+from scipy import signal
 
 logging.basicConfig(level=20)
 
+
 class Audio(object):
     """Streams raw audio from microphone. Data is received in a separate thread, and stored in a buffer, to be read from."""
 
     FORMAT = pyaudio.paInt16
-    RATE = 16000
+    # Network/VAD rate-space
+    RATE_PROCESS = 16000
     CHANNELS = 1
     BLOCKS_PER_SECOND = 50
-    BLOCK_SIZE = int(RATE / float(BLOCKS_PER_SECOND))
 
-    def __init__(self, callback=None):
+    def __init__(self, callback=None, device=None, input_rate=RATE_PROCESS):
         def proxy_callback(in_data, frame_count, time_info, status):
             callback(in_data)
             return (None, pyaudio.paContinue)
-        if callback is None: callback = lambda in_data: self.buffer_queue.put(in_data)
+        if callback is None: 
+            callback = lambda in_data: self.buffer_queue.put(in_data)
         self.buffer_queue = queue.Queue()
-        self.sample_rate = self.RATE
-        self.block_size = self.BLOCK_SIZE
+        self.device = device
+        self.input_rate = input_rate
+        self.sample_rate = self.RATE_PROCESS
+        self.block_size = int(self.RATE_PROCESS / float(self.BLOCKS_PER_SECOND))
+        self.block_size_input = int(self.input_rate / float(self.BLOCKS_PER_SECOND))
         self.pa = pyaudio.PyAudio()
-        self.stream = self.pa.open(format=self.FORMAT,
-                                   channels=self.CHANNELS,
-                                   rate=self.sample_rate,
-                                   input=True,
-                                   frames_per_buffer=self.block_size,
-                                   stream_callback=proxy_callback)
+
+        kwargs = {
+            'format': self.FORMAT,
+            'channels': self.CHANNELS,
+            'rate': self.input_rate,
+            'input': True,
+            'frames_per_buffer': self.block_size_input,
+            'stream_callback': proxy_callback,
+        }
+
+        # if not default device
+        if self.device:
+            kwargs['input_device_index'] = self.device
+
+        self.stream = self.pa.open(**kwargs)
         self.stream.start_stream()
 
+    def resample(self, data, input_rate):
```

Comment:
How does it compared WRT relying on `sox` as in `convert_samplerate()` from `native_client/python/client.py` ?

Reply:
Hi @lissyx, 
I have plotted the audio signal using matplotlib (not sure how to attach images here) and compared the following:
- original 44100 sound clip
- downsampled 16000 with scipy.signal
- downsampled 16000 with sox

Actually, scipy signal is indistinguishable from the original 44100, I have compared the graphs very closely and the 16100 values are visually identical.
Sox however does change the signal a little bit where peaks seem to be lost or smoothed out.

Reply:
So, if scipy.signal is a better way to go, perhaps we should consider this in client.py as well?

Reply:
@jorxster no, pulling `scipy` with the deepspeech modules was introducing a lot of other headaches :)

Reply:
Thanks @lissyx, so scipy here is OK then? I will leave this as scipy unless you have objections, as it is currently working for me like it is.